File: smart-travel-planner/src/main/java/ibu/aisi/smart_travel_planner/core/model/Trip.java
package ibu.aisi.smart_travel_planner.core.model;

import jakarta.persistence.*;

import lombok.*;

import java.util.Date;

@Entity
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "trips")
public class Trip {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String description;

    @Column(name = "start_date")
    private Date startDate;

    @Column(name = "end_date")
    private Date endDate;
    private String status; // "Upcoming" or "Completed"

    @ManyToOne
    @JoinColumn(nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(nullable = false)
    private City city;
}

File: smart_travel_planner_web/src/types/Trip.ts
import { City } from "./City";

export type Trip = {
    id: number;           // Unique identifier for the trip
    name: string;         // Name of the trip
    description: string;  // Description of the trip
    startDate: string;    // Start date of the trip (ISO 8601 format)
    endDate: string;      // End date of the trip (ISO 8601 format)
    status: string;       // Status of the trip (e.g., "Upcoming", "Completed")
    city: City;           // Associated city object
    cityId?: number;      // Optional property for handling form data
    userId?: number;      // Optional property for handling form data
}